sdk_contrib/aws_v1_support/src/test/java/io/opentelemetry/sdk/contrib/trace/aws/propagation/AWSXRayPropagatorTest.java

```diff
@@ -0,0 +1,278 @@
+
+  @Test
+  public void inject_SampledContext() {
+    Map<String, String> carrier = new LinkedHashMap<>();
+    xrayPropagator.inject(
+        withSpanContext(
+            SpanContext.create(TRACE_ID, SPAN_ID, SAMPLED_TRACE_FLAG, TRACE_STATE_DEFAULT),
+            Context.current()),
+        carrier,
+        setter);
+
+    assertThat(carrier)
+        .containsEntry(
+            TRACE_HEADER_KEY,
+            generateTraceHeaderValue(
+                TRACE_ID_BASE16, SPAN_ID_BASE16, "1"));
+  }
+
+  @Test
+  public void inject_SampledContext_nullCarrierUsage() {
+    final Map<String, String> carrier = new LinkedHashMap<>();
+
+    xrayPropagator.inject(
+        withSpanContext(
+            SpanContext.create(TRACE_ID, SPAN_ID, SAMPLED_TRACE_FLAG, TRACE_STATE_DEFAULT),
+            Context.current()),
+        null,
+        new Setter<Map<String, String>>() {
+          @Override
+          public void set(Map<String, String> ignored, String key, String value) {
+            carrier.put(key, value);
+          }
+        });
+
+    assertThat(carrier)
+        .containsEntry(
+            TRACE_HEADER_KEY,
+            generateTraceHeaderValue(
+                TRACE_ID_BASE16, SPAN_ID_BASE16, "1"));
+  }
+
+  @Test
+  public void inject_NotSampledContext() {
+    Map<String, String> carrier = new LinkedHashMap<>();
+    xrayPropagator.inject(
+        withSpanContext(
+            SpanContext.create(TRACE_ID, SPAN_ID, TraceFlags.getDefault(), TRACE_STATE_DEFAULT),
+            Context.current()),
+        carrier,
+        setter);
+    assertThat(carrier)
+        .containsEntry(
+            TRACE_HEADER_KEY,
+            generateTraceHeaderValue(
+                TRACE_ID_BASE16, SPAN_ID_BASE16, "0"));
+  }
+
+  @Test
+  public void extract_EmptyHeaderValue() {
+    Map<String, String> invalidHeaders = new LinkedHashMap<>();
+    invalidHeaders.put(TRACE_HEADER_KEY, "");
+
+    assertThat(getSpanContext(xrayPropagator.extract(Context.current(), invalidHeaders, getter)))
+        .isSameInstanceAs(SpanContext.getInvalid());
+  }
+
+  @Test
+  public void extract_NotEnoughParts() {
+    Map<String, String> invalidHeaders = new LinkedHashMap<>();
+    invalidHeaders.put(TRACE_HEADER_KEY, "aa:bb:cc");
+
+    assertThat(getSpanContext(xrayPropagator.extract(Context.current(), invalidHeaders, getter)))
+        .isSameInstanceAs(SpanContext.getInvalid());
+  }
+
+  @Test
+  public void extract_TooManyParts() {
+    Map<String, String> invalidHeaders = new LinkedHashMap<>();
+    invalidHeaders.put(TRACE_HEADER_KEY, "aa:bb:cc:dd:ee");
+
+    assertThat(getSpanContext(xrayPropagator.extract(Context.current(), invalidHeaders, getter)))
+        .isSameInstanceAs(SpanContext.getInvalid());
+  }
+
+  @Test
+  public void extract_InvalidTraceId() {
+    Map<String, String> invalidHeaders = new LinkedHashMap<>();
+    invalidHeaders.put(
+        TRACE_HEADER_KEY,
+        generateTraceHeaderValue(
```

Comment:
Here especially, don't use a helper method or even a constant, inlining the header will make much more readable what the test is checking.

Reply:
will fix in rev2